topeft/modules/datacard_tools.py

```diff
@@ -422,7 +422,7 @@ def read(self,fpath):
         for km_dist, h in self.hists.items():
             if h.empty():
                 continue
-            if self.var_lst and km_dist not in self.var_lst:
+            if self.var_lst and km_dist not in self.var_lst and km_dist.replace("_sumw2", "") not in self.var_lst:
```

Comment:
I'm not a big fan of having a boolean check do multiple things in order to perform the actual comparison you want. Specifically in this case I would prefer calling `km_dist.replace('_sumw2',''')` before the `if` clause and save it to a reasonably named temporary variable, which is then used in `if` block.

Reply:
Done.